src/components/list-item/list-item.tsx

```diff
@@ -0,0 +1,400 @@
+import {
+    Component,
+    Host,
+    Prop,
+    h,
+    Event,
+    EventEmitter,
+    Element,
+} from '@stencil/core';
+import { getIconName } from '../icon/get-icon-props';
+import type { IconSize } from '../icon/icon.types';
+import { createRandomString } from '../../util/random-string';
+import { ListItem as ListItemInterface } from '../list/list-item.types';
+import { MenuItem } from '../menu/menu.types';
+import { ListSeparator } from '../../global/shared-types/separator.types';
+// Removed legacy Enter-to-click helper; we handle keyboard ourselves now
+import { CheckboxTemplate } from '../checkbox/checkbox.template';
+import { RadioButtonTemplate } from '../radio-button-group/radio-button.template';
+
+/**
+ * This components displays the list item.
+ * This centralizes styles and functionality, and helps reduce redundant code
+ * in consumer components such as `limel-list` and `limel-menu-list`.
+ * @exampleComponent limel-example-list-item-basic
+ * @exampleComponent limel-example-list-item-icon
+ * @exampleComponent limel-example-list-item-icon-size
+ * @exampleComponent limel-example-list-item-pictures
+ * @exampleComponent limel-example-list-item-multiple-lines
+ * @exampleComponent limel-example-list-item-interactive
+ * @exampleComponent limel-example-list-item-radio
+ * @exampleComponent limel-example-list-item-checkbox
+ * @exampleComponent limel-example-list-item-actions
+ * @exampleComponent limel-example-list-item-primary-component
+ * @exampleComponent limel-example-list-item-command-text
+ * @private
+ */
+@Component({
+    tag: 'limel-list-item',
+    shadow: true,
+    styleUrl: 'list-item.scss',
+})
+export class ListItem implements ListItemInterface {
+    /** @inheritDoc */
+    @Prop()
```

Comment:
running `npm run api:update` returns this error:
```
api-extractor 7.52.11  - https://api-extractor.com/

Using configuration from ./api-extractor.json
Using custom TSDoc config from /Users/[my-user-name]/src/lime-elements/tsdoc.json
Analysis will use the bundled TypeScript version 5.8.2

ERROR: Internal Error: Unable to analyze the export "ListItemInterface" in
/Users/[my-user-name]/src/lime-elements/dist/types/components/list/list-item.types.d.ts

You have encountered a software defect. Please consider reporting the issue to the maintainers of this application.
```
and I couldn't figure out what causes it.

In this PR, there are two new `@private` components added, and also an unused prop removed. So we need to update the API signature accordingly.

Reply:
perfect. Tank you so much 🤗 
I'll also add a small fixup after this, for the docs blocks.